pkg/blocks/loader.go

```diff
@@ -121,29 +122,26 @@ func LoadTTP(ttpFilePath string, fsys afero.Fs, execCfg *TTPExecutionConfig, arg
 	case *afero.OsFs:
 		absPath, err := filepath.Abs(ttpFilePath)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		ttp.WorkDir = filepath.Dir(absPath)
 	default:
 		wd, err := os.Getwd()
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		ttp.WorkDir = wd
 	}
-
-	// TODO: refactor directory handling - this is in-elegant
-	// but has less bugs than previous way
-	for _, step := range ttp.Steps {
-		step.SetDir(ttp.WorkDir)
-		if cleanups := step.GetCleanup(); cleanups != nil {
-			for _, c := range cleanups {
-				c.SetDir(ttp.WorkDir)
-			}
-		}
+	execCtx := &TTPExecutionContext{
```

Comment:
execCtx construction was moved in here so that we can call `Validate(...)` below - now every TTP you load will be guaranteed to be validated